src/web/metrics.rs

```diff
@@ -72,29 +78,64 @@ lazy_static::lazy_static! {
     )
     .unwrap();
 
-    pub static ref CONCURRENT_DB_CONNECTIONS: IntGauge = register_int_gauge!(
-        "docsrs_db_connections",
-        "The number of currently used database connections"
+    pub static ref USED_DB_CONNECTIONS: IntGauge = register_int_gauge!(
+        "docsrs_used_db_connections",
+        "The number of used database connections"
+    )
+    .unwrap();
+
+    pub static ref IDLE_DB_CONNECTIONS: IntGauge = register_int_gauge!(
+        "docsrs_idle_db_connections",
+        "The number of idle database connections"
+    )
+    .unwrap();
+
+    pub static ref OPEN_FILE_DESCRIPTORS: IntGauge = register_int_gauge!(
+        "docsrs_open_file_descriptors",
+        "The number of currently opened file descriptors"
+    )
+    .unwrap();
+
+    pub static ref CURRENTLY_RUNNING_THREADS: IntGauge = register_int_gauge!(
+        "docsrs_running_threads",
+        "The number of threads being used by docs.rs"
     )
     .unwrap();
 }
 
 pub fn metrics_handler(req: &mut Request) -> IronResult<Response> {
-    let conn = extension!(req, Pool).get()?;
-
-    QUEUED_CRATES_COUNT.set(
-        ctry!(conn.query("SELECT COUNT(*) FROM queue WHERE attempt < 5;", &[]))
-            .get(0)
-            .get(0),
-    );
-    FAILED_CRATES_COUNT.set(
-        ctry!(conn.query("SELECT COUNT(*) FROM queue WHERE attempt >= 5;", &[]))
-            .get(0)
-            .get(0),
-    );
-
-    let pool = extension!(req, Pool);
-    CONCURRENT_DB_CONNECTIONS.set(pool.connections() as i64);
+    {
```

Comment:
This new scope doesn't seem necessary.

Reply:
It minimizes the time that a db connection is being used, which is something we should be doing more often

Reply:
Gotcha, can you add a comment to that effect?